### DISCUSSION TITLE
Milvus takes a lot of time when i try to insert data one be one

### DISCUSSION THREAD

**Discussion by MUsman1992** [milvus-io/milvus#30678] [Q&A and General discussion]
I have a python application that reads data from the Kafka and stores it in Milvus database. As my code that I have written in Python reads each message from Kafka one by one, when I try to insert this data in Milvus one by one it takes few minutes to store up to 5000 records. Can anyone tell me how can i improve the performance when I try to insert data in Milvus. Should i try to do some kind of batch processing and try to execute multiple insert statements at once or is there any other way to achieve better performance? Can anyone guide me in this regard? Thanks

**Comment by xiaofan-luan** [comment]
you should batch 1000 record to insert one time.
or you could concurrently write.
The former is recommended
